_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -28,7 +28,7 @@ The rule applies to any HTML element that is a [descendant](https://dom.spec.wha
 
 ## Expectation
 
-Each target element is either the row or column header for cells that is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) of the same element as the target element.
+The target element is either the row or column header for cells that is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) of the same [table](https://www.w3.org/TR/wai-aria-1.1/#table) or [grid](https://www.w3.org/TR/wai-aria-1.1/#grid) element, as the target element.
```

Comment:
I still don't think that "this element is the row header for this cell" is properly defined. And I think it gets even worse for column headers.

The algorithm for matching in HTML `table` elements is somewhat tricky (because of colspan and rowspan). I guess it is very similar for pure "role" tables. This also gets tricky when there are multiple overlapping headers or just when the `header` attribute is used.

Typically:
```html
<table>
  <tr> <th id="foo">Foo</th> <th id="bar">Bar</th> </tr>
  <tr> <td>0</td> </tr>
  <tr> <td headers="bar">1 </td> </tr>
</table>
```
should be passing the rule, but
```html
<table>
  <tr> <th id="foo">Foo</th> <th id="bar">Bar</th> </tr>
  <tr> <td>0</td> <td headers="foo">1 </td> </tr>
</table>
```
should be failing it.

Then, it might be enough to add a note that assigning headers cells to data cell is done as per algorithm something (the one currently in the background).